recipes/readline/all/conanfile.py

```diff
@@ -40,56 +42,54 @@ def requirements(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
     def validate(self):
-        if self.settings.compiler == "Visual Studio":
+        if is_msvc(self):
             raise ConanInvalidConfiguration("readline does not support Visual Studio")
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
+    def generate(self):
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
```

Comment:
What is the motivation behind:
* generating the virtualrunenv only when not cross-building
* exposing it to the build context
?

Does the configure step try to run executables in some of the host dependencies? What happens when one is actually cross building?

Reply:
I've suggested this change, otherwise it fails. It's a common pattern and it's documented in autotools template.

Reply:
Thanks!